lib/auth/v4/createCanonicalRequest.js

```diff
@@ -16,13 +17,35 @@ function createCanonicalRequest(params) {
     const pQuery = params.pQuery;
     const pHeaders = params.pHeaders;
     const pSignedHeaders = params.pSignedHeaders;
-    const payloadChecksum = params.payloadChecksum;
+    const service = params.service;
+
+    let payloadChecksum = params.payloadChecksum;
+
+    if (!payloadChecksum) {
+        if (pHttpVerb === 'GET') {
+            payloadChecksum = crypto.createHash('sha256').update('')
```

Comment:
Can't you just set the hash for an empty string without calculating it each time?